R/api.R

```diff
@@ -16,109 +16,379 @@
+  cat('ID:', x$id, '\n')
+  cat('Created:', as.character(x$created), '\n')
+  cat('Karma:', x$karma,'\n')
+  cat('Submitted: ')
+  if(length(x$submitted) > 5)
+    cat(utils::head(x$submitted),'... \n')
+  else
+    cat(x$submitted, '\n')
+}
+
+
+#' @title Get Hacker News by Id
+#'
+#' @description Retrieves the item corresponding to specified id using Hacker News API
+#'
 #' @param id id of the item that should be retrieved
 #'
+#' @export
+#'
 #' @return item correspoing to the specified id
 #'
+#' @examples
+#'
+#' h_id <- get_max_item_id()
+#' hacker_article <- get_item_by_id(h_id)
+#'
 get_item_by_id <- function(id) {
   request_url <- create_request_url(.base_url(), c("item", id))
-  .send_request(request_url)
+  item <- .send_request(request_url)
+  if(!is.null(item))
+  {
+    class(item) <- 'hackerlist'
+    item$time <- as.POSIXct(item$time, origin="1970-01-01")
+  }
+  item
 }
 
-#' Get user corresponding to the specified username
+
+#' @title Get Hacker News user
 #'
-#' @export
+#' @description Get user corresponding to the specified username
 #'
 #' @param username of the user that should be retrieved
 #'
+#' @export
+#'
 #' @return user corresponding to the specified username
 #'
+#' @examples
+#' user <- get_user_by_username("steveddaniels")
+#' print(user)
 get_user_by_username <- function(username) {
   request_url <- create_request_url(.base_url(), c("user", username))
-  .send_request(request_url)
+  user <- .send_request(request_url)
+  class(user) <- 'hackeruser'
+  user$created <- as.POSIXct(user$created, origin="1970-01-01")
+  user
 }
 
-#' Get the current largest item id
+
+#' @title Hacker News item largest id
+#'
+#' @description Retrives largest item id available via Hacker News API
 #'
 #' @export
 #'
 #' @return current largest item id
+#'
+#' @examples
+#' h_id <- get_max_item_id()
+#'
 get_max_item_id <- function() {
   request_url <- create_request_url(.base_url(), c("maxitem"))
   .send_request(request_url)
 }
 
-#' Get top stories
+#' @title Hacker News top stories ids
+#'
+#' @description Retrives top stories ids using Hacker News API
 #'
 #' @export
 #'
 #' @return list of top stories ids
-get_top_stories <- function() {
+#'
+#' @examples
+#' top_stories_ids <- get_top_stories_ids()
+#'
+get_top_stories_ids <- function() {
   request_url <- create_request_url(.base_url(), c("topstories"))
   .send_request(request_url)
 }
 
-#' Get new stories
+#' @title Hacker News top stories
+#'
+#' @description Retrives top stories using Hacker News API
+#'
+#' @param max_items Maximum number of items to retrive. If max_items = NULL, returns all available stories
+#'
+#' @export
+#'
+#' @return List of top stories
+#'
+#' @examples
+#' \dontrun{
+#' top_stories <- get_top_stories()
+#' top_5_stories <- get_top_stories(5)
+#' }
+get_top_stories <- function(max_items = NULL) {
+  ids_list <- get_top_stories_ids()
+  if (!is.null(max_items))
+  {
+    ids_list <- ids_list[1:min(max_items, length(ids_list))]
```

Comment:
There needs to be some kind of validation for the max_items value. What if someone passes a negative integer or a string? You can use the assert function from utils to handle that